scripts/examples-checker/src/main.rs

```diff
@@ -0,0 +1,168 @@
+//! Runs `forc build` or `forc fmt --check` on Sway projects specified by paths or within Sway `examples` directory.
+//!
+//! NOTE: This expects `forc`, `forc-fmt`, and `cargo` to be available in `PATH`.
+
+use anyhow::Result;
+use clap::{ArgEnum, ArgGroup, Parser};
+use std::{
+    fs,
+    io::{self, Write},
+    path::{Path, PathBuf},
+};
+
+#[derive(Parser)]
+#[clap(name = "examples-checker", about = "Forc Examples Checker")]
+#[clap(group(
+        ArgGroup::new("run").required(true).args(&["paths", "all-examples"])))]
+struct Cli {
+    /// Targets all Sway examples found at the paths listed
+    #[clap(long = "paths", short = 'p', multiple_values = true)]
+    pub paths: Vec<PathBuf>,
+
+    /// Targets all Sway examples under /examples
+    #[clap(long = "all-examples")]
+    pub all_examples: bool,
+
+    /// Forc command to run on examples
+    #[clap(arg_enum)]
+    command_kind: CommandKind,
+}
+
+#[derive(ArgEnum, Clone, PartialEq, Eq)]
+enum CommandKind {
+    Build,
+    Fmt,
+}
+
+impl std::fmt::Display for CommandKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            CommandKind::Build => write!(f, "build"),
+            CommandKind::Fmt => write!(f, "fmt"),
+        }
+    }
+}
+
+// Check if the given directory contains `Forc.toml` at its root.
+fn dir_contains_forc_manifest(path: &Path) -> bool {
+    if let Ok(entries) = fs::read_dir(path) {
+        for entry in entries.flatten() {
+            if entry.path().file_name().and_then(|s| s.to_str()) == Some("Forc.toml") {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+fn get_sway_path() -> PathBuf {
+    let mut curr_path = std::env::current_dir().unwrap();
+    loop {
+        if curr_path.ends_with("sway") {
+            return curr_path;
+        }
+        curr_path = curr_path.parent().unwrap().to_path_buf()
+    }
+}
+
+/// Returns true if command ran successfully, false otherwise.
+fn run_forc_command(path: &Path, cmd_args: &[&str]) -> bool {
+    if !path.is_dir() || !dir_contains_forc_manifest(path) {
+        return false;
+    }
+
+    let output = std::process::Command::new("forc")
+        .args(cmd_args)
+        .arg(&path)
+        .output()
+        .expect("failed to run command for example project");
+
+    if !output.status.success() {
+        io::stdout().write_all(&output.stdout).unwrap();
+        io::stdout().write_all(&output.stderr).unwrap();
+        false
+    } else {
+        true
+    }
+}
+
+fn run_forc_build(path: &Path) -> bool {
+    run_forc_command(path, &["build", "--path"])
+}
+
+fn run_forc_fmt(path: &Path) -> bool {
+    run_forc_command(path, &["fmt", "--check", "--path"])
+}
+
+fn print_summary(summary: &Vec<(PathBuf, bool)>, command_kind: CommandKind) {
```

Comment:
Small tip - there is almost never a case where `&Vec<T>` is preferred over `&[T]` in argument position. This is because the only differentiating behaviour between `Vec<T>` and `[T]` is that `Vec` is growable, however a `Vec` can't be grown behind an immutable reference. In other words, this argument could be `&[(PathBuf, bool)]` as `&Vec<T>` automatically derefs to `&[T]`, and `&[T]` is strictly more flexible (e.g. also supports fixed-size arrays, `Box<[T]>`, and so on).

Reply:
Thanks for this tip - definitely is strictly better!